addons/l10n_pk_edi/models/res_config_settings.py

```diff
@@ -0,0 +1,34 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import models, fields, _
+from odoo.exceptions import RedirectWarning
+
+
+class ResConfigSettings(models.TransientModel):
+    _inherit = "res.config.settings"
+
+    l10n_pk_edi_pos_id = fields.Char("PoS ID", related="company_id.l10n_pk_edi_pos_id", readonly=False)
+    l10n_pk_edi_token = fields.Char("Pakistan EDI Token", related="company_id.l10n_pk_edi_token", readonly=False)
+    # This proxy is only to access FBR APIs from out side of Pakistan for testing purpose.
+    # And will remove before stable release. (Use proxy server of USA and Dubai to access)
+    l10n_pk_edi_proxy = fields.Char("HTTPS Proxy", related="company_id.l10n_pk_edi_proxy", readonly=False)
+
+    def l10n_pk_check_ntn_cnic(self):
+        if not self.l10n_pk_edi_pos_id:
+            raise RedirectWarning(
+                _("Please enter a PoS ID in invoice section of configuration."),
+                self.env.ref('account.action_account_config').id,
+                _('Go to Configuration')
+            )
+
+    def l10n_pk_edi_test(self):
+        self.l10n_pk_check_ntn_cnic()
```

Comment:
```suggestion
    def l10n_pk_edi_test(self):
        if not self.l10n_pk_edi_pos_id:
            raise RedirectWarning(
                _("Please enter a PoS ID in invoice section of configuration."),
                self.env.ref('account.action_account_config').id,
                _('Go to Configuration')
            )
```
The method is not useful much 😬